### PR TITLE
Add support for Symfony Forms (Symfony 3.4)

### PR DESCRIPTION
**Pull request by Stanton**
Viewable at: http://localhost:3001/app/form.php

## Symfony Fields

| Field | Styled | Unit Tested | Added to docs |
| ---- | ---- | ---- | ---- |
| TextType          | ✅ | ✅ | ✅ |
| TextareaType   | ✅ | ✅ | ✅ |
| EmailType        | ✅ | ✅ | ✅ |
| IntegerType     | ✅ | ✅ | ✅ |
| MoneyType      | ✅ | ✅ | ✅ |
| NumberType    | ✅ | ✅ | ✅ |
| PasswordType | ✅ | ✅ | ✅ |
| PercentType    | ✅ | ✅ | ✅ |
| SearchType      | ✅ | ✅ | ✅ |
| UrlType             | ✅ | ✅ | ✅ |
| RangeType       | ✅ | ✅ | ✅ |
| TelType            | ✅ | ✅ | ✅ |
| ColorType        | ✅ |  | ✅ | 
| ChoiceType      | ✅ | ✅ | ✅ |
| EntityType        | ✅ | This is a variation of choiceType using a select element populated with data, we can omit the test for this and rely on choiceType test | |
| CountryType    | ✅ | This is a variation of choiceType using a select element populated with data, we can omit the test for this and rely on choiceType test | ✅ |
| LangugeType   | ✅ | This is a variation of choiceType using a select element populated with data, we can omit the test for this and rely on choiceType test | ✅ |
| LocaleType       | ✅ | This is a variation of choiceType using a select element populated with data, we can omit the test for this and rely on choiceType test | ✅ |
| TimezoneType | ✅ | This is a variation of choiceType using a select element populated with data, we can omit the test for this and rely on choiceType test | ✅ |
| CurrencyType  | ✅ | This is a variation of choiceType using a select element populated with data, we can omit the test for this and rely on choiceType test | ✅ |
| DateType              | ✅ |  | ✅ |
| DateIntervalType | ✅ |  | |
| DateTimeType      | ✅ |  | ✅ |
| TimeType              | ✅ |  | ✅ |
| BirthdayType        | ✅ | | |
| CheckboxType | ✅ | ✅ | ✅ |
| FileType            | ✅ | ✅ | ✅ |
| RadioType        | ✅ | ✅ | ✅ |
| CollectionType | ❌ |  | |
| RepeatedType  | ✅ |  | |
| HiddenType | ✅ |  | ✅ |
| ButtonType | ✅ | ✅ | |
| ResetType   | ✅ |  | |
| SubmitType | ✅ |  | |

## Custom Pulsar Fields

| Field | Styled | Unit Tested | Added to docs |
| ---- | ---- | ---- | ---- |
| ToggleSwitchType          | ✅ | ✅ | |
| ChoiceBlockType          | | | |
| ChoiceButtonType          | | | |
| Select2Type          | | | |
| ContentType          | | | |
| DropzoneType          | | | |

### CODE REVIEW

**Comment by jackbentley** [comment]
Once this is merged, we should be able to provide guidance on using Symfony forms for filter bars. This will make implementations more standardised server-side and hopefully simplify their usage

**Comment by jackbentley** [comment]
This PR is for the changes required in pulsar to fully support the Symfony forms styling. See https://github.com/jadu/pulsar-symfony for everything else

File: js/PulsarFormComponent.js [hunk 1]
```diff
@@ -153,14 +154,18 @@ PulsarFormComponent.prototype.initColourpickers = function () {
 
         // changing the picker should update the input
         $pickerInput.spectrum({
-            color: '#' + $input.val(),
+            color: (includePrefix ? '' : '#') + $input.val(),
```
Comment (jackbentley) [inline js/PulsarFormComponent.js:157]: Is this the wrong way around?

Comment (Stanton) [reply js/PulsarFormComponent.js:157]: Smells like it yeah

**Pull request closed without merging**